### PR TITLE
test: add e2e test for config route with plugin orchestration

### LINKED ISSUE #637: add e2e test for config route with plugin orchestration
detail:

 - [x]  add test case for config route with wrong data type for plugin orchestration 

 - [x]  add test case for config route with invalid plugin orchestration

 - [x]  add test case for config route with correct  plugin orchestration 

 - [x]  verify the route just configured 

For background and description, please see:

https://github.com/apache/apisix-dashboard/issues/607

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test

- Related issues

close #637

### CODE REVIEW

**Comment by idbeta** [comment]
`api/test/testdata/dag-conf.json` 
Is it better to put it in this directory `/api/test/e2e/testdata/`?

File: api/test/e2e/route_with_plugin_orchestration_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		{
+			caseDesc:     "make sure the route is not created",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+		},
+		{
+			caseDesc:     "create route with invalid dag conf",
```
Comment (membphis) [inline api/test/e2e/route_with_plugin_orchestration_test.go:45]: what means `conf`? Avoid abbreviations.

File: api/test/e2e/route_with_plugin_orchestration_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			caseDesc:     "make sure the route created failed",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "create route with correct dag cof",
```
Comment (membphis) [inline api/test/e2e/route_with_plugin_orchestration_test.go:63]: what means `cof`?

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_plugin_orchestration_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_orchestration_test.go]: fixed.

File: api/test/e2e/route_with_plugin_orchestration_test.go (hunk 2 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_orchestration_test.go]: fixed.

File: api/test/e2e/route_with_plugin_orchestration_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRoute_With_Plugin_Orchestration(t *testing.T) {
+	bytes, err := ioutil.ReadFile("../testdata/dag-conf.json")
+	assert.Nil(t, err)
+	dagConf := string(bytes)
+	bytes, err = ioutil.ReadFile("../testdata/invalid-dag-conf.json")
```
Comment (gxthrj) [inline api/test/e2e/route_with_plugin_orchestration_test.go:31]: Should we point out where it is invalid ? Or we can comment it.

**Comment by gxthrj** [comment]
CI FAILED

**Comment by nic-chen** [comment]
> `api/test/testdata/dag-conf.json`
> Is it better to put it in this directory `/api/test/e2e/testdata/`?

I think `api/test/testdata/dag-conf.json` is ok.

File: api/test/e2e/route_with_plugin_orchestration_test.go (hunk 3 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_orchestration_test.go:33]: added a comment.

**Comment by nic-chen** [comment]
> CI FAILED

it's ok now.

File: api/test/testdata/dag-conf.json [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+            "customData":{
+              "name":"code == 403",
+              "type":1
+            }
+          },
+          "size":{
+            "height":96,
+            "width":214
+          },
+          "type":"判断条件"
```
Comment (membphis) [inline api/test/testdata/dag-conf.json:157]: Chinese?

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/testdata/dag-conf.json (hunk 4 above)
Comment (nic-chen) [reply api/test/testdata/dag-conf.json]: fixed.

**Review by membphis — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**